build-system/pr-check/cross-browser-tests.js

```diff
@@ -54,6 +54,28 @@ function runIntegrationTestsForPlatform() {
   }
 }
 
+/**
+ * Helper that runs platform-specific E2E tests
+ */
+function runE2eTestsForPlatform() {
+  switch (process.platform) {
+    case 'linux':
+      timedExecOrDie('gulp e2e --nobuild --compiled --browsers=firefox');
+      break;
+    case 'darwin':
+      timedExecOrDie('gulp e2e --nobuild --compiled --browsers=safari');
```

Comment:
To check the really fast run times, can you add `--testnames` to see which tests run?

Reply:
Ran these two steps locally on Safari and noticed that all 165 tests were being flagged as passing before the Safari window opens. Definitely something wrong here. I'll wait for @rileyajones to comment, since he may know something about this that I don't.

```sh
gulp dist --fortesting
gulp e2e --nobuild --compiled --testnames --browsers=safari
```

Reply:
And the results from running with `--testnames` on GH Actions seems to confirm my suspicion: 160+ tests are running and "passing" [in 13 seconds](https://github.com/ampproject/amphtml/pull/32443/checks?check_run_id=1840244246#step:5:544).

Reply:
Debugged some more and figured it out. These are unit tests for the E2E framework. On Safari, they're run first because very few actual E2E tests have opted in, so there's no browser window. On Chrome, other tests (alphabetically earlier than `ControllerPromise`) run first and open the browser. Mystery solved, will merge this PR once tests are green. 

Also, ignore me, I'm talking to the <img width="30" src="https://user-images.githubusercontent.com/26553114/107076250-2ad2c980-67b9-11eb-98da-ae994534376c.png">